HopsanGUI/GUIObjects/GUIModelObject.cpp

```diff
@@ -1824,6 +1838,13 @@ QGraphicsSvgItem *ModelObject::getIcon()
     return mpIcon;
 }
 
+void ModelObject::setAlwaysVisible(const bool visible)
+{
+    mAlwaysVisible = visible;
+
+    mpIcon->setVisible(!mpParentSystemObject->areSignalsHidden());
```

Comment:
Should this be plain and simply set to "true" in this case.
I did not realize, but you can actually interact with the components when they are hidden. (But that is not a problem). So when I check setAlwaysVisible in hidden state I would expect the icon to show up.

Reply:
Not simply to true, but it should be visible if signals are not hidden _or_ if the visibility setting is true.